packages/micropip/src/micropip/package.py

```diff
@@ -0,0 +1,71 @@
+from collections.abc import Collection
+from dataclasses import dataclass, field, astuple
+from pathlib import Path
+from typing import List, Dict, Iterable
+
+__all__: List[str] = []
+
+
+def _format_table(headers: List[str], table: List[Iterable]) -> str:
+    def format_row(values, widths, filler=""):
+        columns = " | ".join(f"{x:{filler}<{w}}" for x, w in zip(values, widths))
+        return f"| {columns} |"
+
+    col_width = [max(len(x) for x in col) for col in zip(headers, *table)]
+    rows = []
+
+    rows.append(format_row(headers, col_width))
+    rows.append(format_row([""] * len(col_width), col_width, filler="-"))
+    for line in table:
+        rows.append(format_row(line, col_width))
+
+    return "\n".join(rows)
+
+
+@dataclass
+class PackageMetadata:
+    name: str
+    version: str = ""
+    source: str = ""
+
+    def __iter__(self):
+        return iter(astuple(self))
+
+    @staticmethod
+    def keys():
+        return [key.capitalize() for key in PackageMetadata.__dataclass_fields__.keys()]
+
+
+class PackageList(Collection):
```

Comment:
Since the expected bahavior is that of a custom dict, I wonder if it wouldn't be simpler to inherit from 
`collections.UserDict` which will make it unnecessary to implement most of the following `__` methods 

Also because it's returned by the list API, we might need to add it to sphinx documentation. Which means it would need a short docstring.